feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts

```diff
@@ -163,21 +174,45 @@ export class ConfigureCartEntryComponent {
    */
   getResolveIssuesA11yDescription(): string | undefined {
     const errorMsgId = 'cx-error-msg-' + this.cartEntry.entryNumber;
-    return !this.readOnly && this.msgBanner ? errorMsgId : undefined;
+    return !this.getDisplayOnly() && this.msgBanner ? errorMsgId : undefined;
   }
 
   /**
-   * Compiles query parameters for the router link. 'resolveIssues' is only set if the component is
+   * Compiles query parameters for the router link.
+   * 'resolveIssues' is only set if the component is
    * rendered in the context of the message banner, and if issues exist at all
+   * 'navigateToCart' is set to 'true' if the navigation to the cart is relevant, otherwise set to 'false'
+   *
    * @returns Query parameters
    */
   getQueryParams(): Params {
     return {
       forceReload: true,
       resolveIssues: this.msgBanner && this.hasIssues(),
+      navigateToCart: this.isCartRelevant(),
     };
   }
 
+  protected isCartRelevant(): boolean {
+    let isCartRelevant = false;
+    const locations = ['cart', 'product', 'search'];
+    this.routingService
+      .getRouterState()
+      .pipe(
+        map((routerState) => {
+          let isFound = false;
+          locations.forEach((location) => {
+            if (routerState.state.semanticRoute === location) {
+              isFound = true;
+            }
+          });
+          return isFound;
+        })
+      )
+      .subscribe((value) => (isCartRelevant = value));
+    return isCartRelevant;
```

Comment:
Here you cannot be sure that the subscription actually happened. Please return an observable instead, then you can get rid of the of the subscription at all